CustomSkillForDataIngestion/QnAIntegrationCustomSkill/Search.cs

```diff
@@ -122,10 +122,13 @@ public static class Search
             output.count = response.Value.TotalCount;
             output.results = response.Value.GetResults().ToList();
             output.facets = facets;
-
+            
```

Comment:
I think in qnamaker there is a way to override source name for a file. If we use that then the source will point to the correct document.

Reply:
I had thought of putting the path to the blob in the source override field but then realized while generating sas token we need the document id from the azure search index, so need to query index again. So I used the source name to find the corresponding document in the search results which will have the document id.

Reply:
understood.